drivers/usb/device/usb_dc_stm32.c

```diff
@@ -338,6 +345,9 @@ static int usb_dc_stm32_init(void)
 #endif /* DT_COMPAT_ST_STM32_USBPHYC */
 	usb_dc_stm32_state.pcd.Init.ep0_mps = USB_OTG_MAX_EP0_SIZE;
 	usb_dc_stm32_state.pcd.Init.vbus_sensing_enable = DISABLE;
+#ifdef CONFIG_USB_DEVICE_SOF
+	usb_dc_stm32_state.pcd.Init.Sof_enable = 1;
```

Comment:
We could enable it also for SoCs with USB controller?

Reply:
@ydamigos Good point. I have move that lines outside of the USB/OTG #ifdef. I have tested that on a nucleo l432kc board.